packages/charting/src/components/HorizontalBarChart/HorizontalBarChart.base.tsx

```diff
@@ -15,43 +15,45 @@ const getClassNames = classNamesFunction<IHorizontalBarChartStyleProps, IHorizon
 type numericAxis = D3Axis<number | { valueOf(): number }>;
 type stringAxis = D3Axis<string>;
 
-@customizable('HorizontalBarChart', ['theme', 'styles'])
 export class HorizontalBarChartBase extends React.Component<IHorizontalBarChartProps, {}> {
-  private _points: IDataPoint[];
-  private _width: number;
-  private _height: number;
-  private _barHeight: number;
-  private _yAxisTickCount: number;
   private _colors: string[];
   private _classNames: IClassNames<IHorizontalBarChartStyles>;
 
+  public static defaultProps: IHorizontalBarChartProps = {
+    data: [],
+    width: 600,
+    height: 350,
+    barHeight: 15,
+    yAxisTickCount: 10
+  };
   constructor(props: IHorizontalBarChartProps) {
     super(props);
 
-    this._points = this.props.data || [];
-
-    this._width = this.props.width || 600;
-    this._height = this.props.height || 350;
-    this._barHeight = this.props.barHeight || 15;
-    this._yAxisTickCount = this.props.yAxisTickCount || 10;
-
     const { theme } = this.props;
     const { palette } = theme!;
     this._colors = this.props.colors || [palette.blueLight, palette.blue, palette.blueMid, palette.blueDark];
   }
 
   public render(): JSX.Element {
-    const isNumeric = typeof this._points[0].x === 'number';
-
-    const xAxis = this._createNumericXAxis(isNumeric);
-    const yAxis = isNumeric ? this._createYAxis() : this._createStringYAxis();
-    const bars = isNumeric ? this._createNumericBars() : this._createStringBars();
+    const points = this.props.data || [];
+    const width = this.props.width || 600;
+    const height = this.props.height || 350;
+    const barHeight = this.props.barHeight || 15;
+    const yAxisTickCount = this.props.yAxisTickCount || 10;
```

Comment:
You shouldn't need these variables at all.. now that you've defined a static `defaultProps` class member, React will automatically populate missing props values for you. You should be able to remove these values and use props directly. Destructing them is common practice, like:

```
const { data, width, height, barHeight, yAxisTickCount } = this.props;
```